components/function-controller/internal/registry/registry_test.go

```diff
@@ -0,0 +1,79 @@
+package registry
+
+import (
+	"context"
+	"net/url"
+	"reflect"
+	"testing"
+)
+
+func TestNewRegistryClient(t *testing.T) {
+	ctx := context.Background()
+
+	tests := []struct {
+		name    string
+		opts    *RegistryClientOptions
+		want    RegistryClient
+		wantErr bool
+	}{
+		{
+			name: "valid options",
+			opts: &RegistryClientOptions{
+				Username: "username",
```

Comment:
You very often use different values for different tests (e.g. "test1", "test2"). In this case you always use "username" and "password". Hypothetically the function being tested might always return "username" and "password" and tests will be green :)